### DISCUSSION TITLE
Lesson 2

### DISCUSSION THREAD

**Discussion by praxis-dev** [smartcontractkit/full-blockchain-solidity-course-py#1396] [Q&A] [answered]
Hi!

In lesson 2 I keep getting this error:

DeclarationError: Identifier not found or not unique.
--> contracts/StorageFactory.sol:8:5:
|
8 | SimpleStorage[] public simpleStorageArray;
| ^^^^^^^^^^^^^

My code:

```
// SPDX-License_Identifier: MIT

pragma solidity ^0.8.0; 

import "./SimpleStorage.sol";

contract StorageFactory {
    SimpleStorage[] public simpleStorageArray;

    function createSimpleStorageContract() public {
        SimpleStorage simpleStorage = new SimpleStorage();
        simpleStorageArray.push(simpleStorage);
    }

}
```

**Comment by cromewar** [comment, marked as answer]
Hello @vaporm you have to use inheritance to make that work:
`
contract StorageFactory is SimpleStorage {}` 
I hope this might help.
